src/compiler/performance.ts

```diff
@@ -1,146 +1,209 @@
 /*@internal*/
 /** Performance measurements for the compiler. */
-namespace ts.performance {
-    let perfHooks: PerformanceHooks | undefined;
-    // when set, indicates the implementation of `Performance` to use for user timing.
-    // when unset, indicates user timing is unavailable or disabled.
-    let performanceImpl: Performance | undefined;
-
-    export interface Timer {
+namespace ts {
+    interface Timer {
         enter(): void;
         exit(): void;
     }
 
-    export function createTimerIf(condition: boolean, measureName: string, startMarkName: string, endMarkName: string) {
-        return condition ? createTimer(measureName, startMarkName, endMarkName) : nullTimer;
+    export interface Statistic {
+        name: string;
+        value: number;
+        type: StatisticType
+    }
+
+    export enum StatisticType {
+        time,
+        count,
+        memory,

```

Comment:
I'm not clear on what this is for, given that we already have a concept of "count" (i.e., "marks"), and "time" (i.e., "measure"). Memory usage isn't generally tracked by the native User Timings API but can be accomplished by [providing additional metadata to a mark](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark). We could probably extend our version of `mark` to support this behavior.